search-commons/src/main/java/no/unit/nva/search/resource/ContributorNodeFilter.java

```diff
@@ -0,0 +1,67 @@
+package no.unit.nva.search.resource;
+
+import static no.unit.nva.constants.Words.AFFILIATIONS;
+import static no.unit.nva.constants.Words.COUNTRY_CODE;
+import static no.unit.nva.constants.Words.ENTITY_DESCRIPTION_CONTRIBUTORS_PATH;
+import static no.unit.nva.constants.Words.IDENTITY;
+import static no.unit.nva.constants.Words.NO;
+import static no.unit.nva.constants.Words.VERIFICATION_STATUS;
+import static no.unit.nva.constants.Words.VERIFIED;
+
+import static nva.commons.core.attempt.Try.attempt;
+
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+
+import no.unit.nva.search.common.records.Midas;
+
+import nva.commons.core.JacocoGenerated;
+
+public class ContributorNodeFilter implements Midas {
+
+    // (package private)
+    static final int FIRST_FIVE_CONTRIBUTORS = 5;
+    static final JsonPointer CONTRIBUTORS_PATH_POINTER = JsonPointer.compile(ENTITY_DESCRIPTION_CONTRIBUTORS_PATH);
+
+    @JacocoGenerated
+    ContributorNodeFilter() {}
+
+    public static ContributorNodeFilter verifiedOrNorwegian() {
+        return new ContributorNodeFilter();
+    }
+
+    @Override
+    public JsonNode transform(JsonNode source) {
+        var target = source.at(CONTRIBUTORS_PATH_POINTER);
+        var elements = target.elements();
+        while (elements.hasNext()) {
+            var element = elements.next();
+            if (deleteAfter(element, FIRST_FIVE_CONTRIBUTORS)
+                    && isNotVerified(element)
+                    && hasNoNorwegianBasedContributor(element)) {
+                elements.remove();
```

Comment:
Instead of deleting everything after 5 - why not take the first 5? This would theoreticly be faster if there are thousands of contriibutors

Reply:
All three statements has to be true, in order to remove an item.